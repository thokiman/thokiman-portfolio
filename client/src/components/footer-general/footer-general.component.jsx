import React from "react";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import {
  FooterContainer,
  FooterCreamPrickInklaim,
  FooterWhiteOutlineThokiman,
  ProfileFooterContainer,
  ProfileFooterIcon,
  ProfileFooterText,
  SocialFooterContainer,
  SocialItem,
} from "./footer-general.styles";
import { selectSocialIconContact } from "../../redux/contact/contact.selectors.js";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors";

const FooterGeneral = ({ socialIcon, isSideBarHidden }) => {
  return (
    <FooterContainer $issidebarhidden={isSideBarHidden}>
      <ProfileFooterContainer>
        <ProfileFooterIcon>
          <FooterCreamPrickInklaim />
          <FooterWhiteOutlineThokiman />
        </ProfileFooterIcon>
        <ProfileFooterText $issidebarhidden={isSideBarHidden}>
          Copyright<sup>&#9400;</sup>2021.
          <em>Allright reserved </em>| Thomas Santosa, B.Eng.
        </ProfileFooterText>
      </ProfileFooterContainer>
      <br />
      <SocialFooterContainer>
        {socialIcon.map(({ iconPath, linkUrl }) => {
          return (
            <SocialItem
              key={iconPath.id}
              to={{ pathname: linkUrl }}
              target="_blank"
              backgroundimage={iconPath}
            />
          );
        })}
      </SocialFooterContainer>
    </FooterContainer>
  );
};

const mapStatetoProps = createStructuredSelector({
  socialIcon: selectSocialIconContact,
  isSideBarHidden: selectIsSideBarHidden,
});

export default connect(mapStatetoProps)(FooterGeneral);
