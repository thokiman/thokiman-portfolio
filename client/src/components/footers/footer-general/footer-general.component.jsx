import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { selectSocialIconContact } from '../../../redux/contact/contact.selectors.js';
import { selectIsSideBarActive } from '../../../redux/header/header.selectors';
import {
  FooterContainer,
  FooterCreamPrickInklaim,
  FooterWhiteOutlineThokiman,
  ProfileFooterContainer,
  ProfileFooterIcon,
  ProfileFooterText,
  SocialFooterContainer,
  SocialItem,
} from './footer-general.styles';

export const FooterGeneral = ({ socialIcon, isSideBarActive }) => {
  return (
    <FooterContainer
      $issidebaractive={isSideBarActive}
      className='footer-general'
    >
      <ProfileFooterContainer>
        <ProfileFooterIcon>
          <div className='footer-general-icon'>
            <FooterCreamPrickInklaim />
          </div>
          <div className='footer-general-icon'>
            <FooterWhiteOutlineThokiman />
          </div>
        </ProfileFooterIcon>
        <ProfileFooterText
          className='footer-text'
          $issidebaractive={isSideBarActive}
        >
          Copyright<sup>&#9400;</sup>2021.
          <em>Allright reserved </em>
          {window.innerWidth <= 800 ? (
            <span>
              <br /> Thomas Santosa, B.Eng.
            </span>
          ) : (
            ' | Thomas Santosa, B.Eng.'
          )}
        </ProfileFooterText>
      </ProfileFooterContainer>
      <br />
      <SocialFooterContainer className='social-footer-container'>
        {socialIcon.map(({ iconPath, linkUrl }) => {
          return (
            <SocialItem
              className='footer-social-icon'
              key={iconPath.id}
              to={{ pathname: linkUrl }}
              target='_blank'
              backgroundimage={iconPath}
            />
          );
        })}
      </SocialFooterContainer>
    </FooterContainer>
  );
};

const mapStatetoProps = createStructuredSelector({
  socialIcon: selectSocialIconContact,
  isSideBarActive: selectIsSideBarActive,
});

export default connect(mapStatetoProps)(FooterGeneral);
