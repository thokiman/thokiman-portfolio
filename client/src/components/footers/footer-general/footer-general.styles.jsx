import { Link } from 'react-router-dom';
import styled, { css, keyframes } from 'styled-components';
import { ReactComponent as CreamPrickInklaim } from '../../../assets/collections-icon/prick-inklaim-icon/logo/prick_inklaim_logo_cream.svg';
import { ReactComponent as WhiteOutlineThokiman } from '../../../assets/collections-icon/thokiman-icon/logo/outline/thokiman_head_logo_cream.svg';

const maxWidthTablet = 1024;
const maxWidthSmartphone = 800;
const blackColor = '#000000';
const heavyLightBlueColor = '#bccfd5';
const whiteSmokeColor = '#e3e1e4';
const greenBlueColor = '#128194';

const FooterIsActivated = css`
  @media only screen and (max-width: ${maxWidthTablet}px) {
    position: relative;
    width: 100vw;
    left: 0vw;
    transition: left 0.5s ease;
    padding: 10px;
    margin: 9px 0;
    height: 18vh;
  }
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    left: -3vw;
  }
  @media only screen and (max-width: 700px) {
    left: 0vw;
  }
`;

const getFooterSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return null;
  return FooterIsActivated;
};
export const FooterContainer = styled.div`
  background-color: ${blackColor};

  width: 103vw;
  height: 14vh;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  position: relative;
  top: 0;
  right: 3vw;
  bottom: 0;
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    position: relative;
    width: 64vw;
    left: 16vw;
    margin: 0.3vw;
    transition: left 0.5s ease;
  }
  @media only screen and (max-width: 550px) {
    left: 29vw;
  }
  @media only screen and (max-width: 500px) {
    left: 32vw;
  }
  @media only screen and (max-width: 400px) {
    left: 36vw;
  }
  @media only screen and (max-width: 350px) {
    left: 33vw;
  }
  @media only screen and (max-width: 300px) {
    left: 37vw;
  }
  ${getFooterSideBarActive}
`;

export const ProfileFooterContainer = styled.div`
  margin-right: 20px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: center;
  width: 60vw;

  @media only screen and (max-width: 340px) {
    display: none;
  }
`;

const piCreamAnimate = keyframes`
                from {
                    stroke: ${heavyLightBlueColor};
                    opacity: 0.6;
                }

                to {
                    stroke: ${whiteSmokeColor};
                    opacity: 1;
                }    

`;

const getWidthProfileFooterIcon = ({ $measureprofilefootericon }) => {
  return $measureprofilefootericon.width;
};

export const ProfileFooterIcon = styled.div`
  width: 200px;
  height: 8vh;
  display: flex;
  justify-content: center;

  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: ${getWidthProfileFooterIcon}vw;
    justify-content: flex-end;
  }

  .thoki-out-cream {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }

  .pi-cream {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }

  .pi-cream-one {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }
`;

const getWidthFooterGeneralIcon = ({ $measurefootergeneralicon }) => {
  return $measurefootergeneralicon.width;
};
const getHeightFooterGeneralIcon = ({ $measurefootergeneralicon }) => {
  return $measurefootergeneralicon.height;
};

export const FooterGeneralIcon = styled.div`
  width: ${getWidthFooterGeneralIcon}vw;
  height: ${getHeightFooterGeneralIcon}vh;
`;
const FooterIcon = css`
  width: 100%;
  height: 100%;
`;

export const FooterCreamPrickInklaim = styled(CreamPrickInklaim)`
  ${FooterIcon}
`;
export const FooterWhiteOutlineThokiman = styled(WhiteOutlineThokiman)`
  ${FooterIcon}
`;

const ProfileFooterTextSideBarActive = css`
  display: none;
`;
const getProfileFooterTextSideBarActive = ({ $issidebaractive }) => {
  if ($issidebaractive) return ProfileFooterTextSideBarActive;
  return null;
};
export const ProfileFooterText = styled.div`
  width: 100%;
  font-family: futura-medium;
  font-kerning: auto;
  letter-spacing: 0.03em;
  text-align: end;
  color: ${whiteSmokeColor};

  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    font-size: 3vw;
    ${getProfileFooterTextSideBarActive}
  }
`;
export const SocialFooterContainer = styled.div`
  width: 170px;
  height: 11vh;
  padding: 0 15px 0;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-auto-rows: 1fr;
  gap: 15px 15px;
  align-items: center;
  justify-items: center;
  border-left: ${whiteSmokeColor} solid 1px;
  @media only screen and (max-width: 350px) {
    grid-template-columns: repeat(3, 37px);
    grid-auto-rows: 23px;
    margin-bottom: 4vh;
    margin-top: 4vh;
  }
  @media only screen and (max-width: 300px) {
    grid-auto-rows: 28px;
  }
`;

const socialItem = keyframes`
                    0%,
                    100% {
                        transform: rotate(0deg);
                        transform-origin: 50% 0;
                        background-color: ${greenBlueColor};
                        border-radius: 5px 5px 5px;
                        opacity: 1;
                    }

                    10% {
                        transform: rotate(2deg);
                        border-radius: 5px 5px 5px;
                        opacity: 0.6;
                    }

                    20%,
                    40%,
                    60% {
                        transform: rotate(-4deg);
                        background-color: ${greenBlueColor};
                        border-radius: 5px 5px 5px;
                        opacity: 1;
                    }

                    30%,
                    50%,
                    70% {
                        transform: rotate(4deg);
                        border-radius: 5px 5px 5px;
                        opacity: 0.5;
                    }

                    80% {
                        transform: rotate(-2deg);
                        background-color: ${greenBlueColor};
                        border-radius: 5px 5px 5px;
                        opacity: 1;
                    }

                    90% {
                        transform: rotate(2deg);
                        border-radius: 5px 5px 5px;
                        opacity: 0.3;
                    }
`;

export const SocialItem = styled(Link)`
  background-repeat: no-repeat;
  height: 35px;
  width: 35px;
  object-fit: cover;

  background-image: url(${({ backgroundimage }) => backgroundimage.iconPath});
  @media only screen and (max-width: ${maxWidthSmartphone}px) {
    width: 6vw;
    height: 4vh;
  }
  @media only screen and (max-width: 400px) {
    width: 7vw;
    height: 5vh;
  }
  @media only screen and (max-width: 400px) and (max-height: 667px) {
    width: 6vw;
    height: 4vh;
  }
  @media only screen and (max-width: 350px) {
    width: 7vw;
    height: 4vh;
  }
  &:hover {
    animation: ${socialItem} 2s ease-in-out infinite;
  }
`;

FooterContainer.displayName = 'FooterContainer';
ProfileFooterContainer.displayName = 'ProfileFooterContainer';
ProfileFooterIcon.displayName = 'ProfileFooterIcon';
FooterCreamPrickInklaim.displayName = 'FooterCreamPrickInklaim';
FooterWhiteOutlineThokiman.displayName = 'FooterWhiteOutlineThokiman';
ProfileFooterText.displayName = 'ProfileFooterText';
SocialFooterContainer.displayName = 'SocialFooterContainer';
SocialItem.displayName = 'SocialItem';
