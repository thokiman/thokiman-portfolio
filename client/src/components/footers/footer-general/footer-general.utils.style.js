import {
  hasScreenSize1,
  hasScreenSize1_1,
  hasScreenSize1_2,
  hasScreenSize1_3,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize3_6,
  hasScreenSize4,
  hasScreenSize5,
  hasScreenSizeWindow,
} from '../../../utils/common.component.utils';

export const measureProfileFooterIconStyles = (viewWidth, viewHeight) => {
  let profileFooterIconProps;

  if (hasScreenSizeWindow(viewWidth, viewHeight)) {
    console.log('profile-footer', 1);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
    console.log('profile-footer', 2);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
    console.log('profile-footer', 3);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
    console.log('profile-footer', 4);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize1(viewWidth)) {
    console.log('profile-footer', 5);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('profile-footer', 6);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('profile-footer', 7);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('profile-footer', 8);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('profile-footer', 9);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('profile-footer', 10);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('profile-footer', 11);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize3_6(viewWidth, viewHeight)) {
    console.log('profile-footer', 12);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('profile-footer', 13);
    profileFooterIconProps = {
      width: 35,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('profile-footer', 14);
    profileFooterIconProps = {
      width: 35,
    };
  } else {
    console.log('profile-footer', 15);
    profileFooterIconProps = {
      width: 35,
    };
  }
  return profileFooterIconProps;
};
export const measureFooterGeneralIconStyles = (viewWidth, viewHeight) => {
  let footerGeneralIconProps;

  if (hasScreenSizeWindow(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 1);
    footerGeneralIconProps = {
      width: 15,
      height: 5,
    };
  } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 2);
    footerGeneralIconProps = {
      width: 10,
      height: 6,
    };
  } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 3);
    footerGeneralIconProps = {
      width: 5,
      height: 7,
    };
  } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 4);
    footerGeneralIconProps = {
      width: 5,
      height: 7,
    };
  } else if (hasScreenSize1(viewWidth)) {
    console.log('footer-general-icon', 5);
    footerGeneralIconProps = {
      width: 11,
      height: 7,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('footer-general-icon', 6);
    footerGeneralIconProps = {
      width: 12,
      height: 7,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 7);
    footerGeneralIconProps = {
      width: 15,
      height: 6,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 8);
    footerGeneralIconProps = {
      width: 16,
      height: 6,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 9);
    footerGeneralIconProps = {
      width: 14,
      height: 6,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 10);
    footerGeneralIconProps = {
      width: 14,
      height: 6,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 11);
    footerGeneralIconProps = {
      width: 15,
      height: 5,
    };
  } else if (hasScreenSize3_6(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 12);
    footerGeneralIconProps = {
      width: 15,
      height: 5,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('footer-general-icon', 13);
    footerGeneralIconProps = {
      width: 5,
      height: 7,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('footer-general-icon', 14);
    footerGeneralIconProps = {
      width: 5,
      height: 7,
    };
  } else {
    console.log('footer-general-icon', 15);
    footerGeneralIconProps = {
      width: 15,
      height: 5,
    };
  }
  return footerGeneralIconProps;
};
