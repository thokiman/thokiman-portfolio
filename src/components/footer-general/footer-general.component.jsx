import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import { selectSocialIconContact } from "../../redux/contact/contact.selectors.js";
import { ReactComponent as CreamPrickInklaim } from "../../assets/collections-icon/prick-inklaim-icon/logo/prick_inklaim_logo_cream.svg";
import { ReactComponent as WhiteOutlineThokiman } from "../../assets/collections-icon/thokiman-icon/logo/outline/thokiman_head_logo_cream.svg";
import "./footer-general.styles.scss";

const FooterGeneral = ({ socialIcon }) => {
  return (
    <div className="footer-container">
      <div className="profile-footer-container">
        <div className="profile-footer-icon">
          <CreamPrickInklaim className="footer-icon" />
          <WhiteOutlineThokiman className="footer-icon" />
        </div>
        <div className="profile-footer-text">
          Copyright<sup>&#9400;</sup>2021.
          <em>Allright reserved </em>| Thomas Santosa, B. Eng.
        </div>
      </div>
      <br />
      <div className="social-footer-container">
        {socialIcon.map(({ iconPath, linkUrl }) => {
          return (
            <Link
              key={iconPath.id}
              to={{ pathname: linkUrl }}
              target="_blank"
              className="social-item"
              style={{
                backgroundImage: `url(${iconPath.iconPath})`,
              }}
            ></Link>
          );
        })}
      </div>
    </div>
  );
};

const mapStatetoProps = createStructuredSelector({
  socialIcon: selectSocialIconContact,
});

export default connect(mapStatetoProps)(FooterGeneral);
