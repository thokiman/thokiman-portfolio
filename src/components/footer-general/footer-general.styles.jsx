import { Link } from "react-router-dom";
import styled, { css, keyframes } from "styled-components";

import { ReactComponent as CreamPrickInklaim } from "../../assets/collections-icon/prick-inklaim-icon/logo/prick_inklaim_logo_cream.svg";
import { ReactComponent as WhiteOutlineThokiman } from "../../assets/collections-icon/thokiman-icon/logo/outline/thokiman_head_logo_cream.svg";
const FooterSideBarActive = css`
  @media only screen and (max-width: 800px) {
    position: relative;
    height: 16vh;
    width: 100vw;
    margin-bottom: 20px;
    left: 0;
    transition: left 0.5s ease;
  }
`;
const getFooterSideBarActive = (props) => {
  if (props.$issidebarhidden) return null;
  return FooterSideBarActive;
};
export const FooterContainer = styled.div`
  background-color: #3f4249;
  height: 14vh;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  @media only screen and (max-width: 800px) {
    position: relative;
    top: 0;
    left: 140px;
    right: 0;
    bottom: 0;
    transition: left 0.5s ease;
    width: 66vw;
  }
  ${getFooterSideBarActive}
`;

export const ProfileFooterContainer = styled.div`
  margin-right: 20px;
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: center;
  width: 60vw;
`;

const piCreamAnimate = keyframes`
                from {
                    stroke: #bccfd5;
                    opacity: 0.6;
                }

                to {
                    stroke: #e3e1e4;
                    opacity: 1;
                }    

`;

export const ProfileFooterIcon = styled.div`
  width: 200px;
  height: 8vh;
  display: flex;
  justify-content: center;
  @media only screen and (max-width: 800px) {
    width: 150px;
  }
  .thoki-out-cream {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }

  .pi-cream {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }

  .pi-cream-one {
    animation: ${piCreamAnimate} 1s ease-in-out infinite;
  }
`;

const FooterIcon = css`
  width: 100%;
  height: 100%;
  margin: 2px;
`;

export const FooterCreamPrickInklaim = styled(CreamPrickInklaim)`
  ${FooterIcon}
`;
export const FooterWhiteOutlineThokiman = styled(WhiteOutlineThokiman)`
  ${FooterIcon}
`;

const ProfileFooterTextSideBarActive = css`
  display: none;
`;
const getProfileFooterTextSideBarActive = (props) => {
  if (props.$issidebarhidden) return ProfileFooterTextSideBarActive;
  return null;
};
export const ProfileFooterText = styled.div`
  width: 100%;
  font-family: futura-medium;
  font-kerning: auto;
  letter-spacing: 0.03em;
  text-align: end;
  color: #e3e1e4;

  @media only screen and (max-width: 800px) {
    font-size: 13px;
    ${getProfileFooterTextSideBarActive}
  }
`;
export const SocialFooterContainer = styled.div`
  width: 170px;
  height: 11vh;
  padding: 0 5px 0;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-auto-rows: 1fr;
  gap: 5px 5px;
  align-items: center;
  justify-items: center;
  border-left: #e3e1e4 solid 1px;
`;

const socialItem = keyframes`
                    0%,
                    100% {
                        transform: rotate(0deg);
                        transform-origin: 50% 0;
                        background-color: #128194;
                        border-radius: 5px 5px 5px;
                        opacity: 1;
                    }

                    10% {
                        transform: rotate(2deg);
                        border-radius: 5px 5px 5px;
                        opacity: 0.6;
                    }

                    20%,
                    40%,
                    60% {
                        transform: rotate(-4deg);
                        background-color: #128194;
                        border-radius: 5px 5px 5px;
                        opacity: 1;
                    }

                    30%,
                    50%,
                    70% {
                        transform: rotate(4deg);
                        border-radius: 5px 5px 5px;
                        opacity: 0.5;
                    }

                    80% {
                        transform: rotate(-2deg);
                        background-color: #128194;
                        border-radius: 5px 5px 5px;
                        opacity: 1;
                    }

                    90% {
                        transform: rotate(2deg);
                        border-radius: 5px 5px 5px;
                        opacity: 0.3;
                    }
`;

export const SocialItem = styled(Link)`
  background-repeat: no-repeat;
  height: 42px;
  width: 42px;
  object-fit: cover;
  background-image: url(${({ backgroundimage }) => backgroundimage.iconPath});
  @media only screen and (max-width: 800px) {
    width: 35px;
    height: 35px;
  }
  &:hover {
    animation: ${socialItem} 2s ease-in-out infinite;
  }
`;

FooterContainer.displayName = "FooterContainer";
ProfileFooterContainer.displayName = "ProfileFooterContainer";
ProfileFooterIcon.displayName = "ProfileFooterIcon";
FooterCreamPrickInklaim.displayName = "FooterCreamPrickInklaim";
FooterWhiteOutlineThokiman.displayName = "FooterWhiteOutlineThokiman";
ProfileFooterText.displayName = "ProfileFooterText";
SocialFooterContainer.displayName = "SocialFooterContainer";
SocialItem.displayName = "SocialItem";
