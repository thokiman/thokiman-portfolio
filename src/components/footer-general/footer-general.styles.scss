.footer-container {
    position: relative;
    top: 10px;
    bottom: 10px;
    right: 10px;
    height: 14vh;

    display: flex;
    justify-content: flex-end;

    .profile-footer-container {

        position: relative;
        right: 30px;
        display: flex;
        flex-direction: column;
        align-items: flex-end;
        justify-content: center;
        width: 30%;
        height: 100%;

        .profile-footer-icon {
            width: 30%;
            height: 50%;

            .thoki-out-cream {
                animation: pi-cream-animate 1s ease-in-out infinite;
            }

            .pi-cream {
                animation: pi-cream-animate 1s ease-in-out infinite;

            }

            .pi-cream-one {
                animation: pi-cream-animate 1s ease-in-out infinite;

            }



            @keyframes pi-cream-animate {
                from {
                    stroke: #bccfd5;
                    opacity: 0.6;
                }

                to {
                    stroke: #e3e1e4;
                    opacity: 1;
                }
            }

            .footer-icon {
                width: 50%;
                height: 90%;

            }
        }

        .profile-footer-text {

            width: 100%;
            font-family: futura-medium;
            font-kerning: auto;
            letter-spacing: 0.03em;
            text-align: end;
            color: #e3e1e4;
        }
    }

    .social-footer-container {

        width: 15%;
        height: 90%;
        display: grid;
        grid-template-columns: repeat(3, 1fr);
        grid-auto-rows: 1fr;
        align-items: center;
        justify-items: center;
        border-left: #e3e1e4 solid 1px;



        .social-item {
            background-repeat: no-repeat;
            height: 80%;
            width: 50%;
            object-fit: cover;




            &:hover {

                animation: shake-top 2s ease-in-out infinite;


                @keyframes shake-top {

                    0%,
                    100% {
                        transform: rotate(0deg);
                        transform-origin: 50% 0;
                        background-color: #128194;
                        border-radius: 5px 5px 5px;
                        opacity: 1;
                    }

                    10% {
                        transform: rotate(2deg);
                        border-radius: 5px 5px 5px;
                        opacity: 0.6;
                    }

                    20%,
                    40%,
                    60% {
                        transform: rotate(-4deg);
                        background-color: #128194;
                        border-radius: 5px 5px 5px;
                        opacity: 1;
                    }

                    30%,
                    50%,
                    70% {
                        transform: rotate(4deg);
                        border-radius: 5px 5px 5px;
                        opacity: 0.5;
                    }

                    80% {
                        transform: rotate(-2deg);
                        background-color: #128194;
                        border-radius: 5px 5px 5px;
                        opacity: 1;
                    }

                    90% {
                        transform: rotate(2deg);
                        border-radius: 5px 5px 5px;
                        opacity: 0.3;
                    }
                }

            }
        }

    }
}