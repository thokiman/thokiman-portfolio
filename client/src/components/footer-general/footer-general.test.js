import React from "react";
import { shallow } from "enzyme";

import { FooterGeneral } from "./footer-general.component";

describe("FooterGeneral component", () => {
  let wrapper;
  const mockSocialIcon = [];
  const mockFalseBool = false;

  beforeEach(() => {
    const mockProps = {
      socialIcon: mockSocialIcon,
      isSideBarHidden: mockFalseBool,
    };
    wrapper = shallow(<FooterGeneral {...mockProps} />);
  });

  it("should render FooterGeneral component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
