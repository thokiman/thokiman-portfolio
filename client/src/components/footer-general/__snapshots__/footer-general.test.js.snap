// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterGeneral component should render FooterGeneral component 1`] = `ShallowWrapper {}`;
